File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int [] arr = new int[2];
        
        for(int i = 0; i < nums.length; i++) {
            
            if (map.containsKey(target - nums[i])) {
                arr[0] = map.get(target - nums[i]);
                arr[1] = i;
                break;
            }
            else {
                map.put(nums[i], i);
            }
        }
        System.out.println(arr[0] + " " + arr[1]);
        return arr;
        
    }
}

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        # compliment + nums[i] = target
        # target - nums[i] = compliment
        
        if len(nums) < 2:
            return []
        
        hashmap = {}
        for i in range(len(nums)):
            compliment = target - nums[i]
            if compliment in hashmap:
                return [hashmap[compliment], i]
            else:
                hashmap[nums[i]] = i;
                
        return []